File: Python/Balanced Binary Tree.py
class Solution:
    # @param root, a tree node
    # @return a boolean
    def depth(self, root):
        if root == None:
            return 0
        else:
            return max(self.depth(root.left), self.depth(root.right)) + 1
    def isBalanced(self, root):
        if root == None:
            return True
        else:
            if self.isBalanced(root.left) and self.isBalanced(root.right):
                return abs(self.depth(root.left) - self.depth(root.right)) <= 1
            else:
                return False

File: C++/Balanced-Binary-Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int helper(TreeNode* root) {
        if (root == NULL) return 0;
        int leftv = helper(root->left), rightv = helper(root->right);
        if (leftv == -1 || rightv == -1 || abs(leftv - rightv) > 1) return -1;
        return max(leftv, rightv) + 1;
    }
    bool isBalanced(TreeNode* root) {
        if (helper(root) == -1) return false;
        return true;
    }
};